modules/phase_field/src/materials/OPInterfaceBarrierMaterial.C

```diff
@@ -0,0 +1,83 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "OPInterfaceBarrierMaterial.h"
+
+template<>
+InputParameters validParams<OPInterfaceBarrierMaterial>()
+{
+  InputParameters params = validParams<Material>();
```

Comment:
Just a suggestion, but it would be helpful to add a class description here using
```
params.addClassDescriptipon("Free Energy contribution that penalizes more than three order parameters being non-zero");
```
```